lib/blocs/activity/activity_cubit.dart

```
@@ -0,0 +1,15 @@
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:meta/meta.dart';
+
+part 'activity_state.dart';
+
+class ActivityCubit extends Cubit<ActivityState> {
+  ActivityCubit() : super(ActivityNotRunning());
+
+  void performUninterruptableActivity(Function activity) async {
+    emit(ActivityInProgress());
```

In the absence of something like a semaphore here to count the number of things in progress, can we please throw an error here if the cubit is already in progress when this function is called?